shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -474,7 +474,8 @@
     overlay_view.get().frame = flutter_view_.get().bounds;
     overlay_view.get().autoresizingMask =
         (UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight);
-    std::unique_ptr<IOSSurface> ios_surface = [overlay_view.get() createSurface:nil];
+    std::unique_ptr<IOSSurface> ios_surface =
```

Comment:
is it possible for an overlay to switch from gl_content to ios_context and vice versa?

Reply:
Not sure I understand. The  IOContextGL is a subclass of IOSContext. Can you clarify?

Reply:
Sorry - I meant if the `EnsureOverlayInitialized` can be called multiple times with the same `overlay_id`, but different values for `gl_context` and `ios_context`. If we are caching these overlays, should that be taken into account?

Reply:
There is no more `gl_context_`, only the ios_context which should be the same reference created and held by the platform view.

Reply:
what about `ios_context` and `gr_context`?

Reply:
The iOS context owns the GrContexts. It owns one for on-screen rendering and another for off-screen rendering. The `gr_context` must be the same for all overlays and be the same as the on-screen context owned by the `IOSContext`.

Reply:
I see. Thanks for the explanation!